internal/k8s/client.go

```diff
@@ -29,21 +27,32 @@ func (cID ContainerID) ShortStr() string {
 	return string(cID)
 }
 
+func (nID NodeID) String() string { return string(nID) }
+
 type Client interface {
 	Apply(ctx context.Context, entities []K8sEntity) error
 	Delete(ctx context.Context, entities []K8sEntity) error
 
 	PodWithImage(ctx context.Context, image reference.NamedTagged) (PodID, error)
 
+	// Gets the ID for the Node on which the specified Pod is running
+	GetNodeForPod(ctx context.Context, podID PodID) (NodeID, error)
+
+	// Finds the PodID for the instance of appName running on the same node as podID
+	FindAppByNode(ctx context.Context, appName string, nodeID NodeID) (PodID, error)
+
 	// Waits for the LoadBalancer to get a publicly available URL,
 	// then opens that URL in a web browser.
 	OpenService(ctx context.Context, lb LoadBalancer) error
 }
 
 type KubectlClient struct {
-	env Env
+	env           Env
+	kubectlRunner kubectlRunner
```

Comment:
since we're mocking at this level, is it now redundant to have a `fakeK8sClient` type? Or are they both useful?

Reply:
I think they're both useful. faking things out via the runner would require tests using a k8s client to know about its implementation (like what output the kubectl process should generate to get the desired result).